### ISSUE TITLE
Missing steps for first installation

### ISSUE THREAD

**Issue by dendory** [greenplum-db/gpdb#171]
So clearly the README has some missing steps when trying to compile and start the database for the first time. After doing a successful 'configure', 'make' and 'make install', the steps go straight to the 'gpdemo' part, but that doesn't work, it says that it cannot connect to the database. After digging around, I found that 'gpstart' is how you start the database engine, but even after setting MASTER_DATA_DIRECTORY to a random folder, it complains about missing files. What step is missing? How do you create a starting database and all related files so we can start the engine?

**Comment by atris** [comment]
I am assuming you did make cluster and that you sourced greenplum_path.sh
before gpstart

**Comment by dendory** [comment]
I sourced it, but 'make cluster' fails for the same reason. It cannot connect to the database on unix socket /tmp/.s.PSQL.15432

**Comment by asimrp** [comment]
Hi @dendory, can you please post details on the error reported by `make cluster`?  And also details about your platform?

**Comment by dendory** [comment]
It's a pretty basic CentOS 7 box. It seems like the make process just can't connect to the database, which makes sense since it isn't running. Is 'make cluster' suppose to set up a database and start it? What exactly is supposed to be setting up the data folder so 'gpstart' can run?

**Comment by asimrp** [comment]
`make cluster` is supposed to initialize a new Greenplum instance with one master, three primary segments and three mirrors.  All these segments are nothing but PostgreSQL instances and their data directories can be found at `gpAux/gpdemo/datadirs`.  At the end of `make cluster`, you should find Greenplum database started with master listening on port 15432.

If you can share the exact output of `make cluster` in your case, it will be easier to identify the problem.

**Comment by andreasscherbaum** [comment]
Yes, "make cluster" is supposed to setup this cluster for you, at least for testing.
We've seen some issues in the past where "make cluster" failed, please post the entire output of the "make cluster" step.

**Comment by dendory** [comment]
[elfguy@centos gpdemo]$ make cluster
======================================================================
            ______  _____  ______  _______ _______  _____
           |  ____ |_____] |     \ |______ |  |  | |     |
           |_____| |       |_____/ |______ |  |  | |_____|

----------------------------------------------------------------------

  This is a demo of the Greenplum Database system.  We will create
  a cluster installation with master and 6 segment instances
  (3 primary & 3 mirror).

    GPHOME ................. : /usr/gpdb
    MASTER_DATA_DIRECTORY .. : /home/elfguy/gpdb/gpAux/gpdemo/datadirs/qddir/demoDataDir-1

    MASTER PORT (PGPORT) ... : 15432
    SEGMENT PORTS .......... : 25432 25433 25434 25435 25436 25437 25438 25439 25440 25441 25442 25443

  NOTE(s):

    * The DB ports identified above must be available for use.
    * An environment file gpdemo-env.sh has been created for your use.

======================================================================

==========================================================================================
executing:
  /usr/gpdb/bin/gpinitsystem -a -c clusterConfigFile ""
==========================================================================================

20151210:21:19:49:002389 gpinitsystem:centos:elfguy-[INFO]:-Checking configuration parameters, please wait...
20151210:21:19:49:002389 gpinitsystem:centos:elfguy-[INFO]:-Reading Greenplum configuration file clusterConfigFile
20151210:21:19:49:002389 gpinitsystem:centos:elfguy-[INFO]:-Locale has not been set in clusterConfigFile, will set to default value
20151210:21:19:49:002389 gpinitsystem:centos:elfguy-[INFO]:-Locale set to en_US.utf8
20151210:21:19:49:002389 gpinitsystem:centos:elfguy-[INFO]:-No DATABASE_NAME set, will exit following template1 updates
20151210:21:19:49:002389 gpinitsystem:centos:elfguy-[INFO]:-MASTER_MAX_CONNECT not set, will set to default value 25
20151210:21:19:50:002389 gpinitsystem:centos:elfguy-[INFO]:-Detected a single host GPDB array build, reducing value of BATCH_DEFAULT from 60 to 4
20151210:21:19:50:002389 gpinitsystem:centos:elfguy-[WARN]:-Master open file limit is 1024 should be >= 65535
20151210:21:19:50:002389 gpinitsystem:centos:elfguy-[INFO]:-Checking configuration parameters, Completed
20151210:21:19:50:002389 gpinitsystem:centos:elfguy-[INFO]:-Commencing multi-home checks, please wait...
.
20151210:21:19:50:002389 gpinitsystem:centos:elfguy-[INFO]:-Configuring build for standard array
20151210:21:19:50:002389 gpinitsystem:centos:elfguy-[INFO]:-Commencing multi-home checks, Completed
20151210:21:19:50:002389 gpinitsystem:centos:elfguy-[INFO]:-Building primary segment instance array, please wait...
...
20151210:21:19:51:002389 gpinitsystem:centos:elfguy-[INFO]:-Building group mirror array type , please wait...
...
20151210:21:19:51:002389 gpinitsystem:centos:elfguy-[INFO]:-Checking Master host
20151210:21:19:51:002389 gpinitsystem:centos:elfguy-[INFO]:-Checking new segment hosts, please wait...
20151210:21:19:52:002389 gpinitsystem:centos:elfguy-[WARN]:-Host centos.seven-kingdoms.local open files limit is 1024 should be >= 65535
......
20151210:21:19:53:002389 gpinitsystem:centos:elfguy-[INFO]:-Checking new segment hosts, Completed
20151210:21:19:53:002389 gpinitsystem:centos:elfguy-[INFO]:-Building the Master instance database, please wait...
20151210:21:22:54:002389 gpinitsystem:centos:elfguy-[INFO]:-Starting the Master in admin mode
elfguy@centos.seven-kingdoms.local's password:
20151210:21:25:21:gpinitsystem:centos:elfguy-[FATAL]:-Hostname lookup for host centos.seven-kingdoms.local failed. Script Exiting!
20151210:21:25:21:002389 gpinitsystem:centos:elfguy-[WARN]:-Script has left Greenplum Database in an incomplete state
20151210:21:25:21:002389 gpinitsystem:centos:elfguy-[WARN]:-Run command /bin/bash /home/elfguy/gpAdminLogs/backout_gpinitsystem_elfguy_20151210_211949 to remove these changes
20151210:21:25:22:002389 gpinitsystem:centos:elfguy-[INFO]:-Start Function BACKOUT_COMMAND
20151210:21:25:22:002389 gpinitsystem:centos:elfguy-[INFO]:-End Function BACKOUT_COMMAND
========================================
gpinitsystem returned: 2
========================================

======================================================================
                           OPTIMIZER STATE
----------------------------------------------------------------------
  Optimizer state .. : psql: FATAL:  System was started in master-only utility mode - only utility mode connections are allowed
======================================================================

psql: FATAL:  System was started in master-only utility mode - only utility mode connections are allowed
======================================================================

make: [cluster] Error 2 (ignored)

[elfguy@centos gpdemo]$ nslookup centos.seven-kingdoms.local
Server:         10.10.106.2
Address:        10.10.106.2#53

Name:   centos.seven-kingdoms.local
Address: 10.10.106.4

[elfguy@centos gpdemo]$ ps ax |grep postgres
 8215 pts/0    S      0:00 /usr/gpdb/bin/postgres -D /home/elfguy/gpdb/gpAux/gpdemo/datadirs/qddir/demoDataDir-1 -i -p 15432 -c gp_role=utility -M master -b 1 -C -1 -z 0 -m
 8216 ?        Ss     0:00 postgres: port 15432, logger process
 8219 ?        Ss     0:00 postgres: port 15432, stats collector process
 8220 ?        Ss     0:00 postgres: port 15432, writer process
 8221 ?        Ss     0:00 postgres: port 15432, checkpoint process
 8222 pts/0    S      0:00 postgres: port 15432, sweeper process
 8729 pts/0    R+     0:00 grep --color=auto postgres
[elfguy@centos gpdemo]$ source gpdemo-env.sh
[elfguy@centos gpdemo]$ gpstop
20151210:21:27:59:008756 gpstop:centos:elfguy-[INFO]:-Starting gpstop with args:
20151210:21:27:59:008756 gpstop:centos:elfguy-[INFO]:-Gathering information and validating the environment...
20151210:21:27:59:008756 gpstop:centos:elfguy-[INFO]:-Obtaining Greenplum Master catalog information
20151210:21:27:59:008756 gpstop:centos:elfguy-[INFO]:-Obtaining Segment details from master...
20151210:21:27:59:008756 gpstop:centos:elfguy-[CRITICAL]:-gpstop failed. (Reason='Database does not contain gp_fault_strategy entry') exiting...
[elfguy@centos gpdemo]$ gpstart
20151210:21:28:03:008765 gpstart:centos:elfguy-[INFO]:-Starting gpstart with args:
20151210:21:28:03:008765 gpstart:centos:elfguy-[INFO]:-Gathering information and validating the environment...
20151210:21:28:03:008765 gpstart:centos:elfguy-[CRITICAL]:-gpstart failed. (Reason='[Errno 2] No such file or directory: '/home/elfguy/gpdb/gpAux/gpdemo/datadirs/qddir/demoDataDir-1/gp_dbid'') exiting...
[elfguy@centos gpdemo]$

**Comment by asimrp** [comment]
Hi @dendory, thank you for sharing the output.

Can you paste the output of `hostname` command on your system?

I also see a password prompt in your output.  `make cluster` needs password-less ssh to be setup so that `ssh localhost` works without prompting for password.

**Comment by dendory** [comment]
Nice to know. Enabled ssh using pre-authorized keys. Another error. This is turning out to be quite a chore :/

		[elfguy@centos gpdemo]$ source /usr/gpdb/greenplum_path.sh
		[elfguy@centos gpdemo]$ make cluster
		======================================================================
		            ______  _____  ______  _______ _______  _____
		           |  ____ |_____] |     \ |______ |  |  | |     |
		           |_____| |       |_____/ |______ |  |  | |_____|

		----------------------------------------------------------------------

		  This is a demo of the Greenplum Database system.  We will create
		  a cluster installation with master and 6 segment instances
		  (3 primary & 3 mirror).

		    GPHOME ................. : /usr/gpdb
		    MASTER_DATA_DIRECTORY .. : /home/elfguy/gpdb/gpAux/gpdemo/datadirs/qddir/demoDataDir-1

		    MASTER PORT (PGPORT) ... : 15432
		    SEGMENT PORTS .......... : 25432 25433 25434 25435 25436 25437 25438 25439 25440 25441 25442 25443

		  NOTE(s):

		    * The DB ports identified above must be available for use.
		    * An environment file gpdemo-env.sh has been created for your use.

		======================================================================

		==========================================================================================
		executing:
		  /usr/gpdb/bin/gpinitsystem -a -c clusterConfigFile ""
		==========================================================================================

		20151210:22:13:50:022674 gpinitsystem:centos:elfguy-[INFO]:-Checking configuration parameters, please wait...
		20151210:22:13:51:022674 gpinitsystem:centos:elfguy-[INFO]:-Reading Greenplum configuration file clusterConfigFile
		20151210:22:13:51:022674 gpinitsystem:centos:elfguy-[INFO]:-Locale has not been set in clusterConfigFile, will set to default value
		20151210:22:13:51:022674 gpinitsystem:centos:elfguy-[INFO]:-Locale set to en_US.utf8
		20151210:22:13:51:022674 gpinitsystem:centos:elfguy-[INFO]:-No DATABASE_NAME set, will exit following template1 updates
		20151210:22:13:51:022674 gpinitsystem:centos:elfguy-[INFO]:-MASTER_MAX_CONNECT not set, will set to default value 25
		20151210:22:13:51:022674 gpinitsystem:centos:elfguy-[INFO]:-Detected a single host GPDB array build, reducing value of BATCH_DEFAULT from 60 to 4
		20151210:22:13:51:022674 gpinitsystem:centos:elfguy-[WARN]:-Master open file limit is 1024 should be >= 65535
		20151210:22:13:51:022674 gpinitsystem:centos:elfguy-[INFO]:-Checking configuration parameters, Completed
		20151210:22:13:51:022674 gpinitsystem:centos:elfguy-[INFO]:-Commencing multi-home checks, please wait...
		.
		20151210:22:13:51:022674 gpinitsystem:centos:elfguy-[INFO]:-Configuring build for standard array
		20151210:22:13:51:022674 gpinitsystem:centos:elfguy-[INFO]:-Commencing multi-home checks, Completed
		20151210:22:13:51:022674 gpinitsystem:centos:elfguy-[INFO]:-Building primary segment instance array, please wait...
		...
		20151210:22:13:52:022674 gpinitsystem:centos:elfguy-[INFO]:-Building group mirror array type , please wait...
		...
		20151210:22:13:53:022674 gpinitsystem:centos:elfguy-[INFO]:-Checking Master host
		20151210:22:13:53:022674 gpinitsystem:centos:elfguy-[INFO]:-Checking new segment hosts, please wait...
		20151210:22:13:53:022674 gpinitsystem:centos:elfguy-[WARN]:-Host centos.seven-kingdoms.local open files limit is 1024 should be >= 65535
		......
		20151210:22:13:54:022674 gpinitsystem:centos:elfguy-[INFO]:-Checking new segment hosts, Completed
		20151210:22:13:54:022674 gpinitsystem:centos:elfguy-[INFO]:-Building the Master instance database, please wait...
		20151210:22:16:58:022674 gpinitsystem:centos:elfguy-[INFO]:-Starting the Master in admin mode
		20151210:22:17:41:022674 gpinitsystem:centos:elfguy-[INFO]:-Commencing parallel build of primary segment instances
		20151210:22:17:41:022674 gpinitsystem:centos:elfguy-[INFO]:-Spawning parallel processes    batch [1], please wait...
		...
		20151210:22:17:41:022674 gpinitsystem:centos:elfguy-[INFO]:-Waiting for parallel processes batch [1], please wait...
		............................................................................................................................................................................................................................................................................................................................................................................
		20151210:22:24:04:022674 gpinitsystem:centos:elfguy-[INFO]:------------------------------------------------
		20151210:22:24:04:022674 gpinitsystem:centos:elfguy-[INFO]:-Parallel process exit status
		20151210:22:24:04:022674 gpinitsystem:centos:elfguy-[INFO]:------------------------------------------------
		20151210:22:24:04:022674 gpinitsystem:centos:elfguy-[INFO]:-Total processes marked as completed           = 3
		20151210:22:24:04:022674 gpinitsystem:centos:elfguy-[INFO]:-Total processes marked as killed              = 0
		20151210:22:24:04:022674 gpinitsystem:centos:elfguy-[INFO]:-Total processes marked as failed              = 0
		20151210:22:24:04:022674 gpinitsystem:centos:elfguy-[INFO]:------------------------------------------------
		20151210:22:24:04:022674 gpinitsystem:centos:elfguy-[INFO]:-Commencing parallel build of mirror segment instances
		20151210:22:24:04:022674 gpinitsystem:centos:elfguy-[INFO]:-Spawning parallel processes    batch [1], please wait...
		...
		20151210:22:24:04:022674 gpinitsystem:centos:elfguy-[INFO]:-Waiting for parallel processes batch [1], please wait...
		.......................
		20151210:22:24:28:022674 gpinitsystem:centos:elfguy-[INFO]:------------------------------------------------
		20151210:22:24:28:022674 gpinitsystem:centos:elfguy-[INFO]:-Parallel process exit status
		20151210:22:24:28:022674 gpinitsystem:centos:elfguy-[INFO]:------------------------------------------------
		20151210:22:24:28:022674 gpinitsystem:centos:elfguy-[INFO]:-Total processes marked as completed           = 3
		20151210:22:24:28:022674 gpinitsystem:centos:elfguy-[INFO]:-Total processes marked as killed              = 0
		20151210:22:24:28:022674 gpinitsystem:centos:elfguy-[INFO]:-Total processes marked as failed              = 0
		20151210:22:24:28:022674 gpinitsystem:centos:elfguy-[INFO]:------------------------------------------------
		20151210:22:24:29:022674 gpinitsystem:centos:elfguy-[INFO]:-Deleting distributed backout files
		20151210:22:24:29:022674 gpinitsystem:centos:elfguy-[INFO]:-Removing back out file
		20151210:22:24:29:022674 gpinitsystem:centos:elfguy-[INFO]:-No errors generated from parallel processes
		20151210:22:24:29:022674 gpinitsystem:centos:elfguy-[INFO]:-Restarting the Greenplum instance in production mode
		20151210:22:24:30:042611 gpstop:centos:elfguy-[INFO]:-Starting gpstop with args: -a -i -m -d /home/elfguy/gpdb/gpAux/gpdemo/datadirs/qddir/demoDataDir-1
		20151210:22:24:30:042611 gpstop:centos:elfguy-[INFO]:-Gathering information and validating the environment...
		20151210:22:24:30:042611 gpstop:centos:elfguy-[INFO]:-Obtaining Greenplum Master catalog information
		20151210:22:24:30:042611 gpstop:centos:elfguy-[INFO]:-Obtaining Segment details from master...
		20151210:22:24:31:042611 gpstop:centos:elfguy-[INFO]:-Greenplum Version: 'postgres (Greenplum Database) 4.3.99.00 build dev'
		20151210:22:24:31:042611 gpstop:centos:elfguy-[INFO]:-There are 0 connections to the database
		20151210:22:24:31:042611 gpstop:centos:elfguy-[INFO]:-Commencing Master instance shutdown with mode='immediate'
		20151210:22:24:31:042611 gpstop:centos:elfguy-[INFO]:-Master host=centos.seven-kingdoms.local
		20151210:22:24:31:042611 gpstop:centos:elfguy-[INFO]:-Commencing Master instance shutdown with mode=immediate
		20151210:22:24:31:042611 gpstop:centos:elfguy-[INFO]:-Master segment instance directory=/home/elfguy/gpdb/gpAux/gpdemo/datadirs/qddir/demoDataDir-1
		20151210:22:24:32:042611 gpstop:centos:elfguy-[INFO]:-Attempting forceful termination of any leftover master process
		20151210:22:24:32:042611 gpstop:centos:elfguy-[INFO]:-Terminating processes for segment /home/elfguy/gpdb/gpAux/gpdemo/datadirs/qddir/demoDataDir-1
		20151210:22:24:32:042698 gpstart:centos:elfguy-[INFO]:-Starting gpstart with args: -a -d /home/elfguy/gpdb/gpAux/gpdemo/datadirs/qddir/demoDataDir-1
		20151210:22:24:32:042698 gpstart:centos:elfguy-[INFO]:-Gathering information and validating the environment...
		20151210:22:24:32:042698 gpstart:centos:elfguy-[INFO]:-Greenplum Binary Version: 'postgres (Greenplum Database) 4.3.99.00 build dev'
		20151210:22:24:32:042698 gpstart:centos:elfguy-[INFO]:-Greenplum Catalog Version: '300701161'
		20151210:22:24:32:042698 gpstart:centos:elfguy-[INFO]:-Starting Master instance in admin mode
		20151210:22:24:34:042698 gpstart:centos:elfguy-[INFO]:-Obtaining Greenplum Master catalog information
		20151210:22:24:34:042698 gpstart:centos:elfguy-[INFO]:-Obtaining Segment details from master...
		20151210:22:24:35:042698 gpstart:centos:elfguy-[INFO]:-Setting new master era
		20151210:22:24:35:042698 gpstart:centos:elfguy-[INFO]:-Master Started...
		20151210:22:24:35:042698 gpstart:centos:elfguy-[INFO]:-Shutting down master
		20151210:22:24:36:042698 gpstart:centos:elfguy-[WARNING]:-Ping to host: 'centos.seven-kingdoms.local' FAILED
		20151210:22:24:36:042698 gpstart:centos:elfguy-[WARNING]:-Ping to host: 'centos.seven-kingdoms.local' FAILED
		20151210:22:24:36:042698 gpstart:centos:elfguy-[WARNING]:-Ping to host: 'centos.seven-kingdoms.local' FAILED
		20151210:22:24:36:042698 gpstart:centos:elfguy-[WARNING]:-Ping to host: 'centos.seven-kingdoms.local' FAILED
		20151210:22:24:36:042698 gpstart:centos:elfguy-[WARNING]:-Ping to host: 'centos.seven-kingdoms.local' FAILED
		20151210:22:24:36:042698 gpstart:centos:elfguy-[WARNING]:-Ping to host: 'centos.seven-kingdoms.local' FAILED
		20151210:22:24:36:042698 gpstart:centos:elfguy-[WARNING]:-Ping to host: 'centos.seven-kingdoms.local' FAILED
		20151210:22:24:36:042698 gpstart:centos:elfguy-[WARNING]:-Skipping startup of segdb on centos.seven-kingdoms.local directory /home/elfguy/gpdb/gpAux/gpdemo/datadirs/dbfast1/demoDataDir0 Ping Failed <<<<<<
		20151210:22:24:36:042698 gpstart:centos:elfguy-[WARNING]:-Skipping startup of segdb on centos.seven-kingdoms.local directory /home/elfguy/gpdb/gpAux/gpdemo/datadirs/dbfast_mirror1/demoDataDir0 Ping Failed <<<<<<
		20151210:22:24:36:042698 gpstart:centos:elfguy-[WARNING]:-Skipping startup of segdb on centos.seven-kingdoms.local directory /home/elfguy/gpdb/gpAux/gpdemo/datadirs/dbfast2/demoDataDir1 Ping Failed <<<<<<
		20151210:22:24:36:042698 gpstart:centos:elfguy-[WARNING]:-Skipping startup of segdb on centos.seven-kingdoms.local directory /home/elfguy/gpdb/gpAux/gpdemo/datadirs/dbfast_mirror2/demoDataDir1 Ping Failed <<<<<<
		20151210:22:24:36:042698 gpstart:centos:elfguy-[WARNING]:-Skipping startup of segdb on centos.seven-kingdoms.local directory /home/elfguy/gpdb/gpAux/gpdemo/datadirs/dbfast3/demoDataDir2 Ping Failed <<<<<<
		20151210:22:24:36:042698 gpstart:centos:elfguy-[WARNING]:-Skipping startup of segdb on centos.seven-kingdoms.local directory /home/elfguy/gpdb/gpAux/gpdemo/datadirs/dbfast_mirror3/demoDataDir2 Ping Failed <<<<<<
		20151210:22:24:36:042698 gpstart:centos:elfguy-[ERROR]:-No segment started for content: 0.
		20151210:22:24:36:042698 gpstart:centos:elfguy-[INFO]:-dumping success segments: []
		20151210:22:24:36:042698 gpstart:centos:elfguy-[INFO]:-----------------------------------------------------
		20151210:22:24:36:042698 gpstart:centos:elfguy-[INFO]:-DBID:2  FAILED  host:'centos.seven-kingdoms.local' datadir:'/home/elfguy/gpdb/gpAux/gpdemo/datadirs/dbfast1/demoDataDir0' with reason:'Failed to Ping on host: centos.seven-kingdoms.local'
		20151210:22:24:36:042698 gpstart:centos:elfguy-[INFO]:-DBID:5  FAILED  host:'centos.seven-kingdoms.local' datadir:'/home/elfguy/gpdb/gpAux/gpdemo/datadirs/dbfast_mirror1/demoDataDir0' with reason:'Failed to Ping on host: centos.seven-kingdoms.local'
		20151210:22:24:36:042698 gpstart:centos:elfguy-[INFO]:-DBID:3  FAILED  host:'centos.seven-kingdoms.local' datadir:'/home/elfguy/gpdb/gpAux/gpdemo/datadirs/dbfast2/demoDataDir1' with reason:'Failed to Ping on host: centos.seven-kingdoms.local'
		20151210:22:24:36:042698 gpstart:centos:elfguy-[INFO]:-DBID:6  FAILED  host:'centos.seven-kingdoms.local' datadir:'/home/elfguy/gpdb/gpAux/gpdemo/datadirs/dbfast_mirror2/demoDataDir1' with reason:'Failed to Ping on host: centos.seven-kingdoms.local'
		20151210:22:24:36:042698 gpstart:centos:elfguy-[INFO]:-DBID:4  FAILED  host:'centos.seven-kingdoms.local' datadir:'/home/elfguy/gpdb/gpAux/gpdemo/datadirs/dbfast3/demoDataDir2' with reason:'Failed to Ping on host: centos.seven-kingdoms.local'
		20151210:22:24:36:042698 gpstart:centos:elfguy-[INFO]:-DBID:7  FAILED  host:'centos.seven-kingdoms.local' datadir:'/home/elfguy/gpdb/gpAux/gpdemo/datadirs/dbfast_mirror3/demoDataDir2' with reason:'Failed to Ping on host: centos.seven-kingdoms.local'
		20151210:22:24:36:042698 gpstart:centos:elfguy-[INFO]:-----------------------------------------------------


		20151210:22:24:36:042698 gpstart:centos:elfguy-[INFO]:-----------------------------------------------------
		20151210:22:24:36:042698 gpstart:centos:elfguy-[INFO]:-   Successful segment starts                                            = 0
		20151210:22:24:36:042698 gpstart:centos:elfguy-[WARNING]:-Failed segment starts                                                = 6   <<<<<<<<
		20151210:22:24:36:042698 gpstart:centos:elfguy-[INFO]:-   Skipped segment starts (segments are marked down in configuration)   = 0
		20151210:22:24:36:042698 gpstart:centos:elfguy-[INFO]:-----------------------------------------------------
		20151210:22:24:36:042698 gpstart:centos:elfguy-[INFO]:-
		20151210:22:24:36:042698 gpstart:centos:elfguy-[INFO]:-Successfully started 0 of 6 segment instances <<<<<<<<
		20151210:22:24:36:042698 gpstart:centos:elfguy-[INFO]:-----------------------------------------------------
		20151210:22:24:36:042698 gpstart:centos:elfguy-[WARNING]:-Segment instance startup failures reported
		20151210:22:24:36:042698 gpstart:centos:elfguy-[WARNING]:-Failed start 6 of 6 segment instances <<<<<<<<
		20151210:22:24:36:042698 gpstart:centos:elfguy-[WARNING]:-Review /home/elfguy/gpAdminLogs/gpstart_20151210.log
		20151210:22:24:36:042698 gpstart:centos:elfguy-[INFO]:-----------------------------------------------------
		20151210:22:24:36:042698 gpstart:centos:elfguy-[INFO]:-Commencing parallel segment instance shutdown, please wait...
		......

		20151210:22:24:47:042698 gpstart:centos:elfguy-[ERROR]:-gpstart error: Do not have enough valid segments to start the array.
		20151210:22:24:52:022674 gpinitsystem:centos:elfguy-[WARN]:
		20151210:22:24:52:022674 gpinitsystem:centos:elfguy-[WARN]:-Failed to start Greenplum instance; review gpstart output to
		20151210:22:24:53:022674 gpinitsystem:centos:elfguy-[WARN]:- determine why gpstart failed and reinitialize cluster after resolving
		20151210:22:24:53:022674 gpinitsystem:centos:elfguy-[WARN]:- issues.  Not all initialization tasks have completed so the cluster
		20151210:22:24:53:022674 gpinitsystem:centos:elfguy-[WARN]:- should not be used.
		20151210:22:24:53:022674 gpinitsystem:centos:elfguy-[WARN]:-gpinitsystem will now try to stop the cluster
		20151210:22:24:53:022674 gpinitsystem:centos:elfguy-[WARN]:
		20151210:22:24:53:043036 gpstop:centos:elfguy-[INFO]:-Starting gpstop with args: -a -i -d /home/elfguy/gpdb/gpAux/gpdemo/datadirs/qddir/demoDataDir-1
		20151210:22:24:53:043036 gpstop:centos:elfguy-[INFO]:-Gathering information and validating the environment...
		20151210:22:24:53:043036 gpstop:centos:elfguy-[ERROR]:-gpstop error: postmaster.pid file does not exist.  is Greenplum instance already stopped?
		20151210:22:24:53:gpinitsystem:centos:elfguy-[WARN]:-Failed to stop new Greenplum instance Script Exiting!
		========================================
		gpinitsystem returned: 2
		========================================

		======================================================================
		                           OPTIMIZER STATE
		----------------------------------------------------------------------
		  Optimizer state .. : psql: could not connect to server: No such file or directory
		        Is the server running locally and accepting
		        connections on Unix domain socket "/tmp/.s.PGSQL.15432"?
		======================================================================

		psql: could not connect to server: No such file or directory
		        Is the server running locally and accepting
		        connections on Unix domain socket "/tmp/.s.PGSQL.15432"?
		======================================================================

		make: [cluster] Error 2 (ignored)

		[elfguy@centos gpdemo]$ ping centos.seven-kingdoms.local
		PING centos.seven-kingdoms.local (10.10.106.4) 56(84) bytes of data.
		64 bytes from centos.seven-kingdoms.local (10.10.106.4): icmp_seq=1 ttl=64 time=0.032 ms
		64 bytes from centos.seven-kingdoms.local (10.10.106.4): icmp_seq=2 ttl=64 time=0.051 ms
		64 bytes from centos.seven-kingdoms.local (10.10.106.4): icmp_seq=3 ttl=64 time=0.059 ms
		^C
		--- centos.seven-kingdoms.local ping statistics ---
		3 packets transmitted, 3 received, 0% packet loss, time 2008ms
		rtt min/avg/max/mdev = 0.032/0.047/0.059/0.012 ms
		[elfguy@centos gpdemo]$ hostname
		centos.seven-kingdoms.local
		[elfguy@centos gpdemo]$

**Comment by dendory** [comment]
Still can't figure this out. Any idea why it's failing to ping the host, when I can ping it fine from the shell?

**Comment by asimrp** [comment]
Hi @dendory we are sorry for the trouble you are going through and appreciate your patience.  Can you please inspect log files in `~/gpAdminlogs` directory?  Especially `gpinitsystem*.log`?  Does it have more details on why the ping failed?  Searching by timestamp `20151210:22:24:36` in the logs may be easier.

**Comment by dendory** [comment]
Nothing else in the logs, the init logs just say to look in the gpstart log, and the gpstart log has what I posted. Seems like it fails to ping the host 7 times then it warns that it can't start any of those segments, and then errors out saying no segment is working.

**Comment by andreasscherbaum** [comment]
How is your hostname defined? In /etc/hosts or via DNS?

**Comment by dendory** [comment]
DNS. Didn't work when I tried with just hosts and hostname file.

**Comment by asimrp** [comment]
If nothing else in logs, we should try running `gpinitsystem` again with log level set to debug.  In order to do this, please make sure there is no `postgres` process running.  Delete contents of data directories under `gpAux/gpdemo/datadirs/*`.  And run `gpinitsystem` manually, not through `make cluster`.  Here are detailed steps, assuming current directory = `gpAux/gpdemo`:

* Kill postgres process if any of them are running
```
killall postgres
ps -ef | grep [p]ostgres
```

* Delete existing Greenplum instance
```
cd datadirs/
rm -rf qddir/* dbfast1/* dbfast2/* dbfast3/* dbfast_mirror1/* dbfast_mirror2/* dbfast_mirror3/*
cd ..
```

* Run `gpinitsystem` in debug mode 
```
gpinitsystem -D -a -c clusterConfigFile ""
```

Note: the last step will generate a lot of output, which we hope, should include clues on why `ping` is failing.

**Comment by dendory** [comment]
We have a winner:

        20151211:18:08:57:064921 gpstart:centos:elfguy-[DEBUG]:-[worker0] got cmd: /bin/ping6 -c 1 centos.seven-kingdoms.local
		20151211:18:08:57:064921 gpstart:centos:elfguy-[DEBUG]:-  ping details: dbid: 2 cmdStr='/bin/ping6 -c 1 centos.seven-kingdoms.local'  had result: cmd had rc=2 completed=True halted=False
		  stdout=''
		  stderr='connect: Invalid argument
		'
		20151211:18:08:57:064921 gpstart:centos:elfguy-[WARNING]:-Ping to host: 'centos.seven-kingdoms.local' FAILED

Why isn't it using ping?

		[elfguy@centos gpdemo]$ /bin/ping6 -c 1 centos.seven-kingdoms.local
		connect: Invalid argument
		[elfguy@centos gpdemo]$ /bin/ping -c 1 centos.seven-kingdoms.local
		PING centos.seven-kingdoms.local (10.10.106.4) 56(84) bytes of data.
		64 bytes from centos.seven-kingdoms.local (10.10.106.4): icmp_seq=1 ttl=64 time=0.034 ms

		--- centos.seven-kingdoms.local ping statistics ---
		1 packets transmitted, 1 received, 0% packet loss, time 0ms
		rtt min/avg/max/mdev = 0.034/0.034/0.034/0.000 ms
		[elfguy@centos gpdemo]$ ls -l /bin/ping
		-rwxr-xr-x 1 root root 44896 Jun 23 18:04 /bin/ping
		[elfguy@centos gpdemo]$ ls -l /bin/ping6
		-rwxr-xr-x 1 root root 49568 Jun 23 18:04 /bin/ping6

**Comment by asimrp** [comment]
Thank you for the details.  I know what's going on on your system.  This is a very specific to your platform.  `ping6` behaves differently on RHEL and OSX as compared to your platform.  The reason `ping6` is being used is because `gpinitsystem` uses `getaddrinfo()` call to get a network interface.  If the network interface is IPV6, (family == `AF_INET6`), then `ping6` is used, otherwise, `ping` is used.

In order to unblock you, I suggest applying the following patch to `gpMgmt/bin/gppylib/commands/unix.py` and running `make install` again.  The patch adds `socket.AF_INET` as argument to `getaddrinfo()` call on line 393. 

```
@@ -390,7 +390,7 @@ class Ping(Command):
         if curr_platform == LINUX or curr_platform == DARWIN:                                                                                                                    
             # Get the family of the address we need to ping.  If it's AF_INET6                                                                                                   
             # we must use ping6 to ping it.                                                                                                                                      
-            addrinfo = socket.getaddrinfo(hostToPing, None)                                                                                                                      
+            addrinfo = socket.getaddrinfo(hostToPing, None, socket.AF_INET)                                                                                                      
             if addrinfo and addrinfo[0] and addrinfo[0][0] == socket.AF_INET6:                                                                                                   
                 pingToUse = SYSTEM.getPing6()                                                                                                                                    
```

Running `make install` after this patch should make `gpinitsystem` use `ping` and not `ping6` on your system.

Note that this is a very specific patch for your platform.  A fix for this issue should be applicable in general.

**Comment by dendory** [comment]
Linked ping to ping6 and it started. This is a fairly basic CentOS 7 host so I'm guessing you're going to run into this a lot.

**Comment by danielgustafsson** [comment]
Closing as this issue has been solved.

**Comment by rahulsingh0978** [comment]
same error on linux ubantu ->

======================================================================
            ______  _____  ______  _______ _______  _____
           |  ____ |_____] |     \ |______ |  |  | |     |
           |_____| |       |_____/ |______ |  |  | |_____|

----------------------------------------------------------------------

  This is a demo of the Greenplum Database system.  We will create
  a cluster installation with master and 6 segment instances
  (3 primary & 3 mirror).

    GPHOME ................. : /usr/local/gpdb
    MASTER_DATA_DIRECTORY .. : /home/poidev/greenplum/gpdb-master/gpAux/gpdemo/datadirs/qddir/demoDataDir-1

    MASTER PORT (PGPORT) ... : 15432
    SEGMENT PORTS .......... : 25432 25433 25434 25435 25436 25437 25438 25439 25440 25441 25442 25443

  NOTE(s):

    * The DB ports identified above must be available for use.
    * An environment file gpdemo-env.sh has been created for your use.

======================================================================

==========================================================================================
executing:
  /usr/local/gpdb/bin/gpinitsystem -a -c clusterConfigFile -l /home/poidev/greenplum/gpdb-master/gpAux/gpdemo/datadirs/gpAdminLogs ""
==========================================================================================

20170922:13:10:24:006766 gpinitsystem:NOIQAUBUNTU02:poidev-[INFO]:-Checking configuration parameters, please wait...
20170922:13:10:25:006766 gpinitsystem:NOIQAUBUNTU02:poidev-[INFO]:-Reading Greenplum configuration file clusterConfigFile
20170922:13:10:25:006766 gpinitsystem:NOIQAUBUNTU02:poidev-[INFO]:-Locale has not been set in clusterConfigFile, will set to default value
20170922:13:10:25:006766 gpinitsystem:NOIQAUBUNTU02:poidev-[INFO]:-Locale set to en_US.utf8
20170922:13:10:25:006766 gpinitsystem:NOIQAUBUNTU02:poidev-[INFO]:-No DATABASE_NAME set, will exit following template1 updates
20170922:13:10:25:006766 gpinitsystem:NOIQAUBUNTU02:poidev-[INFO]:-MASTER_MAX_CONNECT not set, will set to default value 25
20170922:13:10:25:006766 gpinitsystem:NOIQAUBUNTU02:poidev-[INFO]:-Detected a single host GPDB array build, reducing value of BATCH_DEFAULT from 60 to 4
20170922:13:10:25:006766 gpinitsystem:NOIQAUBUNTU02:poidev-[WARN]:-Master open file limit is 1024 should be >= 65535
20170922:13:10:25:006766 gpinitsystem:NOIQAUBUNTU02:poidev-[INFO]:-Checking configuration parameters, Completed
20170922:13:10:25:006766 gpinitsystem:NOIQAUBUNTU02:poidev-[INFO]:-Commencing multi-home checks, please wait...
.
20170922:13:10:25:006766 gpinitsystem:NOIQAUBUNTU02:poidev-[INFO]:-Configuring build for standard array
20170922:13:10:25:006766 gpinitsystem:NOIQAUBUNTU02:poidev-[INFO]:-Commencing multi-home checks, Completed
20170922:13:10:25:006766 gpinitsystem:NOIQAUBUNTU02:poidev-[INFO]:-Building primary segment instance array, please wait...
...
20170922:13:10:26:006766 gpinitsystem:NOIQAUBUNTU02:poidev-[INFO]:-Building group mirror array type , please wait...
...
20170922:13:10:26:006766 gpinitsystem:NOIQAUBUNTU02:poidev-[INFO]:-Checking Master host
20170922:13:10:26:006766 gpinitsystem:NOIQAUBUNTU02:poidev-[WARN]:-Have lock file /tmp/.s.PGSQL.15432.lock but no process running on port 15432
20170922:13:10:26:gpinitsystem:NOIQAUBUNTU02:poidev-[FATAL]:-Found indication of postmaster process on port 15432 on Master host Script Exiting!
========================================
gpinitsystem returned: 2
========================================

======================================================================
                           OPTIMIZER STATE
----------------------------------------------------------------------
  Optimizer state .. : psql: could not connect to server: Connection refused
        Is the server running locally and accepting
        connections on Unix domain socket "/tmp/.s.PGSQL.15432"?
======================================================================

psql: could not connect to server: Connection refused
        Is the server running locally and accepting
        connections on Unix domain socket "/tmp/.s.PGSQL.15432"?
======================================================================

Makefile:48: recipe for target 'cluster' failed
make: *** [cluster] Error 2

**Comment by gaos1** [comment]
@boogabee @danielgustafsson it seems that the issue is still there. I can see the following error on my centos 7 vm when make cluster.
```
20180831:16:10:40:000421 gpstart:sdw:gpadmin-[INFO]:-DBID:2  FAILED  host:'sdw' datadir:'/home/gpadmin/workspace/gpdb/gpAux/gpdemo/datadirs/dbfast1/demoDataDir0' with reason:'Failed to Ping on host: sdw'
20180831:16:10:40:000421 gpstart:sdw:gpadmin-[INFO]:-DBID:3  FAILED  host:'sdw' datadir:'/home/gpadmin/workspace/gpdb/gpAux/gpdemo/datadirs/dbfast2/demoDataDir1' with reason:'Failed to Ping on host: sdw'
```

after applied the patch as Asim's suggestions, I can make cluster successfully.

**Comment by danielgustafsson** [comment]
My Python is quite non-existing - but to my reading that patch cannot be applied as a generic solution though. Can you open a new issue with your output and link to this one from there to keep the history intact?
